tests/update_test_files.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
+
+"""
+This script is designed to update the expected test output.
+It assumes that you have already run nosetests and ran the entire test suite.
+By doing so, the output has been generated under the given outputs directory.
+and that is what will be used to generate the new expected output under `tests/data/experiments`.
+
+#############################################################################################
+# IMPORTANT: DO NOT RUN THIS SCRIPT BEFORE RUNNING THE TEST SUITE OR IT WILL BE DISASTROUS. #
+#############################################################################################
+
+The scripts works as as follows. For each experiment test:
+- The script locates the output under the given outputs directory.
+- New and changed files under `test_outputs` are copied over to the expected test output location.
+- Old files in the expected test output that DO NOT exist under `test_outputs` are deleted.
+- Files that are already in the expected test output and have not changed are left alone.
+
+The script prints a log detailing the changes made for each experiment test.
+
+:author: Anastassia Loukina
+:author: Nitin Madnani
+:date: Feburary 2018
+"""
+
+import argparse
+import re
+import sys
+
+from ast import literal_eval as eval
+from filecmp import dircmp
+from importlib.machinery import SourceFileLoader
+from inspect import getmembers, getsourcelines, isfunction
+from os import remove
+from os.path import dirname, join
+from pathlib import Path
+from shutil import copyfile
+
+_MY_PATH = dirname(__file__)
+
+
+def is_skll_excluded_file(filename):
+
+    stem = Path(filename).stem
+    suffix = Path(filename).suffix
+    return suffix == '.model' or \
+        suffix == '.npy'or \
+        stem.endswith('_postprocessing_params') or \
+        stem.endswith('_eval') or \
+        stem.endswith('_eval_short') or \
+        stem.endswith('_confMatrix') or \
+        stem.endswith('_pred_train') or \
+        stem.endswith('_pred_processed') or \
+        stem.endswith('_score_dist')
+
+
+def update_reference_output(source, outputs_dir, skll=False):
+
+    # locate the outputs for the experiment under the outputs directory
+    # and also locate the existing experiment outputs
+    reference_output_path = outputs_dir / source / "output"
```

Comment:
To me the naming is very confusing as my intuition would be that reference `refers` to `data/tests/experiments/`. Perhaps we can use `reference`  to refer to the files in `data/tests/experiment` and use something like `updated outputs` to refer to new outputs?

Reply:
Isn't "reference" the actual output you want and that would be in `test_outputs`? But I can rename things to be `updated` and `existing`, that probably makes more sense.

Reply:
I could see your logic when I read the code carefully, but it wouldn't have been my intuitive interpretation so I think it's worth making it clearer. `updated` and `existing` is good.